packages/core/src/avatar/internal/useImage.ts

```diff
@@ -0,0 +1,35 @@
+import { ImgHTMLAttributes, useEffect, useState } from "react";
+import { useIsomorphicLayoutEffect } from "../../utils";
+
+export function useImage({ src }: ImgHTMLAttributes<HTMLImageElement>) {
+  const [status, setStatus] = useState<
+    "pending" | "loading" | "loaded" | "error"
+  >("loading");
+
+  useEffect(() => {
+    setStatus(src ? "loading" : "pending");
+  }, [src]);
+
+  useIsomorphicLayoutEffect(() => {
```

Comment:
Changed this to useLayout so the status check happens quicker and the content doesn't flash as bad